src/go/rpk/pkg/cli/topic/config.go

```diff
@@ -76,6 +76,8 @@ valid, but does not apply it.
 			// at the same time, so we issue first the set request for write,
 			// then the rest of the requests.
 			// See https://github.com/redpanda-data/redpanda/issues/9191
+			// TODO: Remove this once v24.2 is EOL.
+			// See https://github.com/redpanda-data/redpanda/pull/23545.
 			_, isRRR := setKVs["redpanda.remote.read"]
```

Comment:
We can't backport this PR because of the feature gate, but if we wanted to fix these bugs in older versions we could implement this request-splitting logic that rpk is doing inside redpanda in 24.1/24.2/23.3. It might be worth asking product if we want to do that.

To be clear, I think we should get this PR in because this seems like the right long-term fix, but since these bugs are important we might want to release a backportable fix for them as well.

Reply:
> We can't backport this PR because of the feature gate, but if we wanted to fix these bugs in older versions we could implement this request-splitting logic that rpk is doing inside redpanda in 24.1/24.2/23.3. It might be worth asking product if we want to do that.


I think it's a good idea, so that it can work with other tools that support setting arbitrary config.

Reply:
Agreed, these are pretty important bugs, it would be good to fix them in previous versions.

This should be done in a future PR, but for now my focus is on `v24.3`.